transport/http/peer.go

```diff
@@ -98,7 +129,6 @@ func (p *httpPeer) MaintainConn() {
 	// Attempt to retain an open connection to each peer so long as it is
 	// retained.
 	for {
-		p.Peer.SetStatus(peer.Connecting)
 		if p.isAvailable() {
 			p.Peer.SetStatus(peer.Available)
```

Comment:
It’s technically fine since Connecting isn’t particularly useful, but I’ve found a way to keep it, by moving it to OnDisconnect and after exponential back-off, to ensure that the state remains steadily Available if we re-enter the loop due to OnSuspect.